cmd/smi-metrics/main.go

```diff
@@ -210,18 +210,29 @@ func run(_ *cobra.Command, args []string) {
 	log.Infof("api listening on %d", viper.GetInt("api-port"))
 	log.Infof("admin listening on %d", viper.GetInt("admin-port"))
 
-	queries := metrics.Queries{
-		ResourceQueries: viper.GetStringMapString("resourceQueries"),
-		EdgeQueries:     viper.GetStringMapString("edgeQueries"),
+	var meshInstance mesh.Mesh
+	var err error
+	if viper.Get("linkerd") != nil {
```

Comment:
You’ll want to fail fast, especially since `log.Fatalf` will end the program.

```
if viper.Get(“linkerd”) == nil {
    log.Fatalf()
}

var config linkerd.Config
….
```